src/timesync/timesyncd-manager.c

```diff
@@ -1130,3 +1137,50 @@ int manager_new(Manager **ret) {
 
         return 0;
 }
+
+static int manager_save_time_handler(sd_event_source *s, uint64_t usec, void *userdata) {
+        Manager *m = userdata;
+
+        assert(m);
+
+        return manager_save_time_and_rearm(m);
+}
+
+int manager_save_time_add(Manager *m) {
+        int r;
+
+        if (m->save_time_interval_usec > 0) {
+                r = sd_event_add_time_relative(
```

Comment:
This timer should also be removed in `manager_free`

Reply:
Duh. Fixed.